master/internal/api_tasks.go

```diff
@@ -429,10 +430,33 @@ func constructTaskLogsFilters(req *apiv1.TaskLogsRequest) ([]api.Filter, error)
 		}
 	}
 
-	addInFilter("allocation_id", req.AllocationIds, len(req.AllocationIds))
-	addInFilter("agent_id", req.AgentIds, len(req.AgentIds))
+	addNumFilter := func(field string, values []string) {
```

Comment:
this seems like "add num filter" and more like "add cleaned in filter"

Reply:
going with `addNullInclusiveFilter`